cloudevents/emitter.go

```diff
@@ -0,0 +1,60 @@
+package cloudevents
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/cloudevents/sdk-go/v2/client"
+	"github.com/cloudevents/sdk-go/v2/event"
+	"github.com/cloudevents/sdk-go/v2/protocol/http"
+	"github.com/cloudevents/sdk-go/v2/types"
+	"github.com/google/uuid"
+)
+
+type Emitter struct {
+	Endpoint    string
+	Source      string
+	Type        string
+	Id          string
+	Data        string
+	ContentType string
+}
+
+func NewEmitter() *Emitter {
+	return &Emitter{
+		Source:      "/boson/fn",
+		Type:        "boson.fn",
+		Id:          uuid.NewString(),
+		Data:        "",
+		ContentType: event.TextPlain,
+	}
+}
+
+func (emitter *Emitter) Emit(endpoint string) (err error) {
+	c, err := newClient(endpoint)
+	if err != nil {
+		return
+	}
+	e := event.Event{
+		Context: event.EventContextV1{
+			Type:   emitter.Type,
+			Source: *types.ParseURIRef(emitter.Source),
+			ID:     emitter.Id,
+		}.AsV1(),
+	}
+	e.SetData(emitter.ContentType, emitter.Data)
+	result := c.Send(context.TODO(), e)
+	if result.Error() != "" {
+		return fmt.Errorf(result.Error())
+	}
+	return nil
+}
+
+func newClient(target string) (c client.Client, err error) {
+	p, err := http.New(http.WithTarget(target))
```

Comment:
`kn-event` already supports public sending via http, and in-cluster sending via sender job (thus buggy https://github.com/knative-sandbox/kn-plugin-event/issues/12).

Reply:
Yes, we have discussed this and want to provide a simple user experience that is tied directly to a user's function project. As `func` replicates many `kn` capabilities but in a simpler way, so too do we want to provide a similar, simple experience with event emission.